core/src/main/java/org/elasticsearch/index/query/IndexQueryParserService.java

```diff
@@ -135,6 +140,11 @@ public void setTermsLookupFetchService(@Nullable  TermsLookupFetchService termsL
         this.termsLookupFetchService = termsLookupFetchService;
     }
 
+    @Inject(optional = true)
```

Comment:
Yeah it seems to be. If I put this in the constructor I get the following exception:
```
org.elasticsearch.common.inject.ConfigurationException: Guice configuration errors:

1) No implementation for org.elasticsearch.client.Client was bound.
  while locating org.elasticsearch.client.Client
    for parameter 0 at org.elasticsearch.index.search.shape.ShapeFetchService.<init>(Unknown Source)
  while locating org.elasticsearch.index.search.shape.ShapeFetchService
    for parameter 12 at org.elasticsearch.index.query.IndexQueryParserService.<init>(Unknown Source)
  while locating org.elasticsearch.index.query.IndexQueryParserService

1 error
	at __randomizedtesting.SeedInfo.seed([7F8BC9EF44EC1932]:0)
	at org.elasticsearch.common.inject.InjectorImpl.getProvider(InjectorImpl.java:856)
	at org.elasticsearch.common.inject.InjectorImpl.getProvider(InjectorImpl.java:795)
	at org.elasticsearch.common.inject.InjectorImpl.getInstance(InjectorImpl.java:867)
	at org.elasticsearch.index.query.AbstractQueryTestCase.init(AbstractQueryTestCase.java:177)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1638)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$4.evaluate(RandomizedRunner.java:787)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:803)
	at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:46)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at org.apache.lucene.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:42)
	at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:39)
	at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:39)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at org.apache.lucene.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:54)
	at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:48)
	at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:65)
	at org.apache.lucene.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:55)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:365)
	at java.lang.Thread.run(Thread.java:745)


```